quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -69,12 +77,18 @@ public class PropertyMapper<T> {
     private final String description;
     private final BooleanSupplier required;
     private final String requiredWhen;
+    private Matcher fromWildcardMatcher;
```

Comment:
Could a subclass be used to encapsulate th state / construction logic? It might even work well for overloading some of the methods.

Reply:
I was thinking about it but if feels like it'd be perhaps an unnecessary boiler plate code. IMHO the matchers and patterns is something that is owned and used only by a property mapper, they are it's properties.